pkg/clusteragent/clusterchecks/dispatcher_configs.go

```diff
@@ -90,29 +90,27 @@ func (d *dispatcher) removeConfig(digest string) {
 
 	node, found := d.store.getNodeStore(d.store.digestToNode[digest])
 
-	checkName := ""
-	if found {
-		node.RLock()
-		checkName = node.digestToConfig[digest].Name
-		node.RUnlock()
-	}
+	checkName := d.store.digestToConfig[digest].Name
 
 	delete(d.store.digestToNode, digest)
 	delete(d.store.digestToConfig, digest)
 	delete(d.store.danglingConfigs, digest)
 
-	for k, v := range d.store.idToDigest {
-		if v == digest {
-			configsInfo.Delete(node.name, checkName, string(k), le.JoinLeaderValue)
-			delete(d.store.idToDigest, k)
+	var checkIDToRemove checkid.ID
+	for checkID, checkDigest := range d.store.idToDigest {
+		if checkDigest == digest {
+			checkIDToRemove = checkID
+			delete(d.store.idToDigest, checkIDToRemove)
 		}
 	}
 
 	// Remove from node configs if assigned
 	if found {
 		node.Lock()
+		nodeName := node.name
 		node.removeConfig(digest)
 		node.Unlock()
+		configsInfo.Delete(nodeName, checkName, string(checkIDToRemove), le.JoinLeaderValue)
```

Comment:
Is it worth checking if we found the check id to remove with an `if checkIDToRemove != ""` ?

Reply:
I'm not sure. I think it should never be empty unless there's a bug somewhere else. Also, there's no harm in calling `Delete` on a Gauge with labels that form a combination that doesn't exist. It will simply do nothing.

Reply:
After the fix mentioned above, the check IDs are now in a list, so it's no longer an issue.